### PR TITLE
Restructure documentation

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
PR that restructures the documentation to take in mind the user's journey #572 

Includes other changes:
* Updating main README documentation (components, some text tweaks, runner visuals, ...)
* Renaming some components to match the format <component_function>_<modality> (e.g. text_normalization -> normalize_text) 

 There are many other changes that need to be made but best to track them in separate PRs to make reviewing them easier. Created issues can be found here #572

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,50 +24,66 @@ It offers:
...
-    🔧 Plug ‘n’ play composable pipelines for creating datasets for
+    🔧 Plug ‘n’ play composable pipelines for creating datasets for:
     <ul>
         <li>AI image generation model fine-tuning (Stable Diffusion, ControlNet)</li>
         <li>Large language model fine-tuning (LLaMA, Falcon)</li>
     </ul>
 </li>
 <li>
-🧱 Library of off-the-shelf reusable components for
+🧱 Off-the-shelf library of easily adaptable reusable components for:
```
Comment (RobbeSneyders) [inline README.md:34]: Nit, but I don't think "off-the-shelf" is used correctly here. The components are off-the-shelf, not the library :slightly_smiling_face:

File: README.md [hunk 2]
```diff
@@ -24,50 +24,66 @@ It offers:
 
 <ul>
 <li>
-    🔧 Plug ‘n’ play composable pipelines for creating datasets for
+    🔧 Plug ‘n’ play composable pipelines for creating datasets for:
     <ul>
         <li>AI image generation model fine-tuning (Stable Diffusion, ControlNet)</li>
         <li>Large language model fine-tuning (LLaMA, Falcon)</li>
```
Comment (RobbeSneyders) [inline README.md:30]: Let's add RAG here already as well.

File: README.md [hunk 3]
```diff
@@ -24,50 +24,66 @@ It offers:
...
+    <ul>
+        <li>Images: cropping, resizing, etc. </li>
+        <li>Text: chunking, indexing, etc. </li>
+    </ul>
+    </li>
+    <li>
+    Data retrieval
+    <ul>
+        <li>Common Crawl</li>
+        <li>LAION</li>
```
Comment (RobbeSneyders) [inline README.md:63]: Can we summarize every category of filtering on a single line? Just so the view is a bit more condensed.

File: README.md [hunk 4]
```diff
@@ -24,50 +24,66 @@ It offers:
...
-    <li>Enriching data (captioning, metadata generation, synthetics, …)</li>
-    <li>Transparency, auditability, compliance</li>
     </ul>
 </li>
 <li>📖 🖼️ 🎞️ ♾️ Out of the box multimodal capabilities: text, images, video, etc.</li>
-<li>🐍 Standardized, Python/Pandas-based way of creating custom components</li>
-<li>🏭 Production-ready, scalable deployment</li>
-<li>☁️ Multi-cloud integrations</li>
+<li>🐍 A simple Pandas based interface for creating custom components</li>
+<li>🏭 Scalable deployment for production environments.</li>
```
Comment (RobbeSneyders) [inline README.md:70]: I would keep `production-ready` here, as it's more of a USP of Fondant than `scalable`.

File: README.md [hunk 5]
```diff
@@ -24,50 +24,66 @@ It offers:
...
-In the age of Foundation Models, control over your data is key and building pipelines
-for large-scale data processing is costly, especially when they require advanced
-machine learning-based operations. This need not be the case, however, if processing
-components would be reusable and exchangeable and pipelines were easily composable.
-Realizing this is the main vision behind Fondant.
+In the era of Foundation Models, preserving data control has become paramount. 
+Yet, the construction of pipelines for extensive data processing remains a costly affair, particularly in the context of 
+complex machine learning-based operations. 
+Fondant's mission is to streamline this process by introducing reusable and interchangeable processing components, 
+thus establishing easily composed pipelines as the new norm.
```
Comment (RobbeSneyders) [inline README.md:80]: I'm not the biggest fan of this rewrite, I think the new text is more complex.

File: README.md [hunk 6]
```diff
@@ -153,22 +178,39 @@ pip install git+https://github.com/ml6team/fondant.git
...
+            </figure>
+        </td>
+        <td width="33.33%" style="border: 0px; width: 33.33%">
+            <figure>
+                <img src="https://github.com/ml6team/fondant/blob/main/docs/art/runners/kubeflow_pipelines.png?raw=true" />
+                <figcaption class="caption"><strong>KubeflowRunner</strong>: Runs on Kubeflow Pipelines</figcaption>
+            </figure>
+        </td>
+    </tr>
+</table>
```
Comment (RobbeSneyders) [inline README.md:213]: I would already add a Sagemaker Runner as well, with some :construction: Under construction :construction:  warning :)

File: docs/caching.md
Comment (RobbeSneyders) [inline docs/caching.md:1]: We start with a very technical explanation of how caching is implemented. Can we start with a high level explanation of why, how, and when to use it?

File: docs/documentation_guide.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+Learn how to use Fondant to build your own data processing pipeline.
+
+-> Design your own fondant [pipeline](pipeline.md) using the Fondant pipeline API.  
+-> Use existing [reusable components](components/hub.md) to build your pipeline.  
+-> Use [generic components](components/generic_component.md) to load/write your custom data format
+to/from Fondant.  
+-> Build your own [custom component](components/custom_component.md) using the Fondant component
+API.
+-> Learn how to publish your own [components](components/publishing_components.md) to a container
```
Comment (RobbeSneyders) [inline docs/documentation_guide.md:23]: I think you need two spaces at the end of the line above to make this one appear on a new line.

File: docs/documentation_guide.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    }
+</style>
+
+-> [LocalRunner](runners/local.md): ideal for developing fondant pipelines and components faster.   
+-> [VertexRunner](runners/vertex.md): used for running a fondant pipeline on Vertex AI.  
+-> [KubeflowRunner](runners/kfp.md): used for running a fondant pipeline on a Kubeflow cluster.
+
+## Fondant Explorer
+
+Learn how to use the Fondant [data explorer](data_explorer.md) to explore the outputs of your
```
Comment (RobbeSneyders) [inline docs/documentation_guide.md:69]: Also intermediate steps :slightly_smiling_face:

File: docs/guides/installation.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+## Installing Fondant
+
+Install Fondant by running:
+
+```
+pip install fondant
+```
```
Comment (RobbeSneyders) [inline docs/guides/installation.md:7]: I would document our extras here as well.

File: docs/runners/kfp_infrastructure.md
Comment (RobbeSneyders) [inline docs/runners/kfp_infrastructure.md:1]: I don't think this file is included in the tree. Not sure if it's needed though, it's linked from the kubeflow runner page and you can find it via search.

File: docs/runners/kfp_infrastructure.md [hunk 10]
```diff
@@ -1,23 +1,31 @@
...
 
 !!! note "IMPORTANT"
-    - These script serve just a kickstart to help you setup Kubeflow for running Fondant, these are not production ready environments.
-    - Spinning up a cluster on a cloud vendor will incur a cost.
-    - You should never run a script without inspecting it so please familiarize yourself with the commands defined in the Makefiles and adapt it to your own needs.
+- These script serve just a kickstart to help you setup Kubeflow for running Fondant, these are not
+production ready environments.
+- Spinning up a cluster on a cloud vendor will incur a cost.
+- You should never run a script without inspecting it so please familiarize yourself with the
+commands defined in the Makefiles and adapt it to your own needs.
```
Comment (RobbeSneyders) [inline docs/runners/kfp_infrastructure.md:22]: This breaks the admonition.

File: docs/runners/kfp_infrastructure.md [hunk 11]
```diff
@@ -1,23 +1,31 @@
...
 
 ## If you already have a kubernetes cluster
- 
-If you already have setup a kubernetes cluster and you have configured kubectl you can install kubeflow pipelines following this [guide](https://www.kubeflow.org/docs/components/pipelines/v1/installation/standalone-deployment/#deploying-kubeflow-pipelines)
 
+If you already have setup a kubernetes cluster and you have configured kubectl you can install
```
Comment (RobbeSneyders) [inline docs/runners/kfp_infrastructure.md:26]:
```suggestion
If you already have a kubernetes cluster  set up and you have configured kubectl you can install
```

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @PhilippeMoussalli! This is gonna be a great improvement!

Can you address my smaller comments in this PR and create tickets for the larger ones? Then we can already merge this PR and continue the work in smaller separated PRs.

I think we can still have a look at some individual pages and see how we can write them more from a user perspective and how they can use it, instead of explaining how it works. I'll run through the documentation again when it's merged and create tickets where I think this is the case.

File: docs/guides/first_pipeline.md [hunk 12]
```diff
@@ -1,26 +1,9 @@
...
-
-Install Fondant by running:
-
-```
-pip install fondant
-```
-
 Clone the Fondant GitHub repository
 
 ```
```
Comment (Hakimovich99) [inline docs/guides/first_pipeline.md:9]: Link at the bottom of the page "[guide](guides/build_a_simple_pipeline.md)." down. Must send to "Building your own pipeline.

**Review by Hakimovich99 — COMMENTED** [review]
This top-bottom approach where we explain first how to build a pipeline before the components looks good to me. Very intuitive. 
I would add in the "documentation guide" under "getting started with Fondant" a bullet point to the guides  (“building your own pipeline” and “implement custom components”). I think it is important to have them. I have pinned in the docs where I think these should be added.
Also, in those guides, we should add more links to the "theory" which are the "Pipeline" and "Components" sections, but that's already a detail. 
I also wrote some other details I have found, mostly regarding broken links, but I haven't checked them all so some might remain. 
Thanks Philippe!!

File: docs/documentation_guide.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+## Getting started with Fondant
+
+Learn about the Fondant project and how to get started with it.
+
+→ Start with the official guide on how to [install](installation.md) Fondant.  
+→ Get started by running your first fondant [pipeline](guides/first_pipeline.md) using the [local
+runner](runners/local.md). There you will learn how to use reusable components and how to build your
+own custom components.  
+→ Learn how to use the [data explorer](data_explorer.md) to explore the outputs of your pipeline.
```
[the remark this answers is not in the document]
Comment (Hakimovich99) [reply docs/documentation_guide.md:11]: (between the second and third bullet point)

[the remark this answers is not in the document]
Comment (PhilippeMoussalli) [reply docs/documentation_guide.md:11]: Good catch, forgot about that one

File: README.md (hunk 2 above)
Comment (PhilippeMoussalli) [reply README.md:30]: Done :)

File: README.md (hunk 6 above)
Comment (PhilippeMoussalli) [reply README.md:217]: Good point, done ;)

File: docs/runners/kfp_infrastructure.md
Comment (PhilippeMoussalli) [reply docs/runners/kfp_infrastructure.md:1]: Added it to the advanced concepts

File: docs/guides/installation.md (hunk 9 above)
Comment (PhilippeMoussalli) [reply docs/guides/installation.md:7]: done!

File: docs/caching.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
+## Caching
```
[the remark this answers is not in the document]
Comment (PhilippeMoussalli) [reply docs/caching.md]: Removed

File: docs/guides/first_pipeline.md (hunk 12 above)
Comment (PhilippeMoussalli) [reply docs/guides/first_pipeline.md:9]: Good catch, thanks

File: docs/caching.md
Comment (PhilippeMoussalli) [reply docs/caching.md:1]: Created a ticket for this #610

**Comment by PhilippeMoussalli** [comment]
@Hakimovich99 @RobbeSneyders 

Thanks for the feedback! I addressed the main points and added 2 tickets for other ones that should be tackled here #572

**Review by RobbeSneyders — APPROVED** [review]
Thanks @PhilippeMoussalli! Huge task, nice to see it done this quickly :)

**Pull request merged**